Added resource input to task modal and aligned buttons

pedroelton opened the pull request:
Hi, I am sending a little video showing what was done and what I would like you guys to check.
Thank you!

4min video explaining: https://drive.google.com/file/d/1P4XUZYBBWihTCraSeBOXm4XvpIOWFp-Y/view?usp=share_link

RafaelCastilhoCastro commented on src/components/Projects/WBS/WBSDetail/AddTask/AddTaskModal.jsx:
```diff
@@ -1,19 +1,47 @@
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/prop-types */
+/* eslint-disable jsx-a11y/scope */
+/* eslint-disable react/no-array-index-key */
```
Why all these commented lines were added? They were added to other files too but I don't see the reason.

pedroelton replied:
It's added by the quick-fix to hide the error from eslint. Most of the errors are not errors, just good practice suggestions by eslint that I can't fix in code that was written by someone else. things like keeping one line of code under 100 characters and others, if I change it I might break the application since it would be necessary even rename variables or functions.

RafaelCastilhoCastro commented:
Hi @pedroelton,
I tested the PR and it looks good. I found some issues though.

1. The user has to click the name of a selected person to deselect them. I suggest adding an "X" right after the name simbolizing the close function.
![error1](https://user-images.githubusercontent.com/65736872/216418906-dbf957f9-ab65-425c-9f4e-8c94c5c398d7.jpg)

2. Alternatively to removing the list from the form, it might be interesting to set a "window box" so the names will be contained inside. So the user can scroll through the names without changing the modal's size.
![error2](https://user-images.githubusercontent.com/65736872/216418955-2403035a-69ad-41f2-b962-c21be818c596.jpg)

3. I didn't understand the purpose for the Paste button in this context. Maybe we can think of a better name dependending of its function?

4. The new resources field requires some input filtering. For now it is accepting numbers and special characters too. Since you're FE only, that might be out of this PR's scope. If that's the case, please check with Jae about adding the task to the bugs doc if it isn't already there.

5. The fields in the Hours section are accepting negative numbers and saving the data like that. Probably that's also out of this PR's scope. So I think it might be good to check with jae about adding this fix to the list too if its not already there.
![error5](https://user-images.githubusercontent.com/65736872/216420023-10301c07-a7dc-4456-ba86-6b895c87c92e.jpg)

6. The Edit task modal isn't displaying your modifications. Will this form be covered in this PR too?


Thanks!

pedroelton commented:
Hey Rafael, first of all, thank you for your feedback! It was great and also helped me a lot here.
Answering your questions:
1 - Great idea, I'm implementing it right now.
![image](https://user-images.githubusercontent.com/72199676/216489581-3038a23e-3cad-481b-b0a6-57ce26364591.png)
2 - The idea is to change this layout, this strange plus button, and an auto-completion filter. Jay wants to keep those icons, I will see if I can keep them instead of the blue badges I made.
![image](https://user-images.githubusercontent.com/72199676/216491063-3eda2ba4-7b2e-4c34-8c13-563c74620422.png)
3 - I also didn't get it, there's no send button. I just changed the order of the buttons and made the reset red, to be more clear it's a destructive action. But I also didn't understand the paste button, I will ask Jae about it. I was actually supposed to just work on the modal, and the other changes I suggested and Jae liked the changes.
4 - Yes, I'm FE only, I'm studying BE as well, but still not familiar with it at all. My background is more towards UX/UI design and web design, started studying development around 1 year ago.
5 - Yes, I didn't do anything with this component other than just change the position and redesign it. I will write two bugs/fixes, one for this, also another one addressing the new component I made, to add real data.

> Hi @pedroelton, I tested the PR and it looks good. I found some issues though.
> 
> 1. The user has to click the name of a selected person to deselect them. I suggest adding an "X" right after the name simbolizing the close function.
>    ![error1](https://user-images.githubusercontent.com/65736872/216418906-dbf957f9-ab65-425c-9f4e-8c94c5c398d7.jpg)
> 2. Alternatively to removing the list from the form, it might be interesting to set a "window box" so the names will be contained inside. So the user can scroll through the names without changing the modal's size.
>    ![error2](https://user-images.githubusercontent.com/65736872/216418955-2403035a-69ad-41f2-b962-c21be818c596.jpg)
> 3. I didn't understand the purpose for the Paste button in this context. Maybe we can think of a better name dependending of its function?
> 4. The new resources field requires some input filtering. For now it is accepting numbers and special characters too. Since you're FE only, that might be out of this PR's scope. If that's the case, please check with Jae about adding the task to the bugs doc if it isn't already there.
> 5. The fields in the Hours section are accepting negative numbers and saving the data like that. Probably that's also out of this PR's scope. So I think it might be good to check with jae about adding this fix to the list too if its not already there.
>    ![error5](https://user-images.githubusercontent.com/65736872/216420023-10301c07-a7dc-4456-ba86-6b895c87c92e.jpg)
> 6. The Edit task modal isn't displaying your modifications. Will this form be covered in this PR too?
> 
> Thanks!

AnishPandita commented:
Hey @pedroelton I verified your PR and tested out the changes for the frontend issue #672. The padding for the buttons on the the main project page looks great. For the create task popup the resources text box seems to work fine with feature to remove the selected person. For the name dropdown the dropdown menu doesn't seem to close if opened once. The Assigned and Status options look okay too. For the hours option there seems to be some bug when i increment values in best case tab the worst case text box seems to increase in size. Rest everything looks great.

https://user-images.githubusercontent.com/111321945/216711713-85d4ab66-65c1-4bae-9b0b-7529c5848f10.mov

PrabhjotSembhi commented:
Hey, @pedroelton I am not seeing changes. I have tried multiple times. Do you have any idea what the issue can be? 
I am in the current branch. 
![PR 672 2](https://user-images.githubusercontent.com/82662768/216787770-f5ea4916-5a50-41af-88d1-043fad073acd.jpg)
![PR 672 1](https://user-images.githubusercontent.com/82662768/216787773-007d0fa4-24d9-4f84-87c1-6c14a312bbf3.jpg)

pedroelton commented:
Hi Prabhjot, it's still not merged yet, you can do `git checkout pedro_add_task_moda` I guess it might work. Maybe you are looking at the developer branch. :)


> Hey, @pedroelton I am not seeing changes. I have tried multiple times. Do you have any idea what the issue can be? I am in the current branch. ![PR 672 2](https://user-images.githubusercontent.com/82662768/216787770-f5ea4916-5a50-41af-88d1-043fad073acd.jpg) ![PR 672 1](https://user-images.githubusercontent.com/82662768/216787773-007d0fa4-24d9-4f84-87c1-6c14a312bbf3.jpg)

PrabhjotSembhi commented:
Hey @pedroelton, I am getting this compilation error material UI Icon is not found.
I am in this "pedro_add_task_modal".

![image](https://user-images.githubusercontent.com/82662768/216804022-71687387-0be1-4b4e-ac60-828f00cf1ac7.png)

pedroelton commented:
If you want to see with the icon, you can install npm install `@mui/icons-material @mui/material @emotion/styled @emotion/react --force` but I think I'm removing it, some people had the same problem.

pedroelton commented:
I just changed the icon, now you might be able to test it well. The icon that I used was Material UI Icon, but the package wasn't installed in the main repo, I just saw that the main repo had FontAwesome, so I updated it.

Filipe-Oliv commented:
Hi @pedroelton .

I tested it and it's working like in your video, but I encountered the same issue as @AnishPandita encountered.

[week 03 - 02-07-23 (4).webm](https://user-images.githubusercontent.com/114119668/217350793-eb7a0f09-d539-4d63-b38a-787ab1208990.webm)

NyahJ commented:
Hi @pedroelton,

Thanks for your efforts on this amazing redesign work, it makes this feature more functional and understandable for users!

Here I have some advice which I believe could make this work even better:

1. A tiny issue, is it better for the format of "Hours"(red-circled) to be aligned with the other titles(green-underlined)?
![截屏2023-02-07 下午1 04 00](https://user-images.githubusercontent.com/77643813/217365879-411528a1-e091-4224-895f-6d59997b4b1e.png)

2.00:00 - 02:36 of the video: The selection logic of the "Resources" part works as a dropdown list, but the implementation is not, which is little bit confusing and weird. Is it possible for us to make it a real dropdown list?

3.02:36 - 03:12 of the video: Is it possible for the old panel to be updated to the new version?

4.03:12 - 04:10 of the video: The "Copy" function does not work with so many errors, could you pls take a look at it?

https://user-images.githubusercontent.com/77643813/217376751-f7725847-2c54-4f6f-9032-d69b0cbe9551.mp4

pedroelton commented:
Hi @Filipe-Oliv, about this Hour issue, I'm going to check now, I didn't make this component, just changed place to restyle. Maybe it caused that issue.

> Hi @pedroelton .
> 
> I tested it and it's working like in your video, but I encountered the same issue as @AnishPandita encountered related to the hours option. In case you open a new PR for that issue I'll approve this PR.
> 
>  week.03.-.02-07-23.4.webm

Theophileaseh reviewed (commented):
## Suggestions

Hi @pedroelton 

Great work on implementing the features.
I think a dropdown selection will be better for the resources section if one is to add just a single resource or you can make use of checkboxes if one is to select multiple resources. ♻️

one-community commented:
We have that now @Theophileaseh, this PR is to replace that with Pedro's creation because it is more functional.

one-community commented:
Pedro, please resolve the branch conflicts too

Gukal7 commented:
Hi @pedroelton, I have tested your PR and the changes are working fine. But like @AnishPandita mentioned there are some fixes still to be made like for the name dropdown menu not closing once opened. Also the box sizes for hours increases as we increment values. 
Apart from these I have noticed the Task loading box appears in yellow but there is no padding between that and the Refresh button. Also, whenever I click the Add Task button, I have noticed that the loading time is more.

https://user-images.githubusercontent.com/30505274/218417902-873da5a4-abda-48b5-af3a-fe5fef114df8.mov

pedroelton commented:
Hi Gukal, that dropdown from the old component will not be used anymore. About the hour input size changing, I will have a look at it on Thursday, when I'm back. My kid is sick, so these days are pretty busy here. :)

> Hi @pedroelton, I have tested your PR and the changes are working fine. But like @AnishPandita mentioned there are some fixes still to be made like for the name dropdown menu not closing once opened. Also the box sizes for hours increases as we increment values. Apart from these I have noticed the Task loading box appears in yellow but there is no padding between that and the Refresh button. Also, whenever I click the Add Task button, I have noticed that the loading time is more.
> 
>  Screen.Recording.2023-02-13.at.2.26.27.PM.mov

vitorsaa2k commented:
Hello @pedroelton, i have some suggestions about the UX of the TagsSearch component

I think it should work like showed in the video below:

https://user-images.githubusercontent.com/17365161/218835064-f95d6a48-ad11-4059-ae6d-d4cc814c02b2.mp4  

1. Make the dropdown disapear when the input is not on focus and appear again when focused.

2. Make the input value controlled.

3. Make the input value be reseted to an empty string when a name is selected.   

This last suggestion is just a "if you wanna add" differently from the other ones which are "must have"

4. Also, the filter could search from the first letters to the last letters of each name, for example: when you search for "pe" it should show the names that starts with "pe" for example "pedro". Will be needed the build a better algorithm to deal with this searching, but it should not be too hard.

Lunaliar commented:
Hey @pedroelton, I just spent some time testing out your feature and I think it's a definite improvement and a lot more user-friendly. 
It ran fine when I tried to use it a few times besides a couple of layout issues.
![Screenshot from 2023-02-16 10-42-08](https://user-images.githubusercontent.com/98275089/219423752-5488e4fe-0e7c-4c98-a38f-bf9b89636e0c.png)
Here I ran into a slight issue where if I removed the value from the best-case input the worst-case would grow in width.

![Screenshot from 2023-02-16 11-06-13](https://user-images.githubusercontent.com/98275089/219426649-3b1e7565-825b-4192-a3c6-92f196f187f4.png)

I also took a look at TagSearch,jsx and had a few questions about your state "selectedIndex". It seems to be only used in one place to toggle styling for list item on line 76. Which I thought made sense if there's a future plan but because it never gets changed it is basically a constant 0 value. Then I took a closer look at the classname and realized its the same thing even if the boolean is true or false. 

Also I believe the list items need keys to help React keep track of the elements in that array.

Great work!

PrabhjotSembhi commented on src/components/Projects/WBS/WBSDetail/AddTask/components/TagsSearch.jsx:
```diff
@@ -0,0 +1,104 @@
+    setIsHidden(!isHidden);
+  };
+
+  const handleFilter = event => {
```
My suggestions for handling change function

```
const [searchResource, setSearchResource] = useState("")

  const handleFilter = (event) => {
    const searchResourceStr = event.target.value.toLocaleLowerCase();
    setSearchResource(searchResourceStr);
  };
```

PrabhjotSembhi commented on src/components/Projects/WBS/WBSDetail/AddTask/components/TagsSearch.jsx:
```diff
@@ -0,0 +1,104 @@
+  const handleFilter = event => {
+    const searchWord = event.target.value;
+    const noRepeatTags = data.filter(e => !tags.includes(e.name));
+    const newFilter = noRepeatTags.filter(value =>
```
I think this will work. 
```
 
array.filter((element) => {
          return array.name.toLocaleLowerCase().includes(searchResource);
        })
```

if this doesn't work by just plugging in existing code you can figure it out.

PrabhjotSembhi reviewed (commented):
Hey @pedroelton, Filter is not working properly please do the required changes and I will approve this.

Please check the video 

https://user-images.githubusercontent.com/82662768/219444498-7f8ecb60-4d69-44ed-872f-a6d6563bd44d.mp4

Jorge-Nerd commented:
@pedroelton  It seems to me to have the problems that @NyahJ  pointed out when adding the members, the rest seems fine !

suigunfjc commented on src/components/Projects/WBS/WBSDetail/AddTask/components/TagsSearch.jsx:
```diff
@@ -0,0 +1,104 @@
+            placeholder={placeholder}
+            className="border border-dark rounded form-control px-2"
+            onKeyUp={e => (e.key === 'Enter' ? addTags(e) : null)}
+            onChange={handleFilter}
```
This can make the dropdown disappear when not focusing on the input.
`onBlur={e => setIsHidden(true)}`

pedroelton replied:
Thank you, it works. But the problem is, it also hides the tags.

suigunfjc reviewed (commented):
Hi @pedroelton, I tested your functions and I think you've done a great job. These features are very helpful for users. 

I agreed with @PrabhjotSembhi 's suggestions on handling filter input. Storing it in a state can make it easier for preprocessing and validation.

I also made a small change to realize the first suggestion from @vitorsaa2k to remove the dropdown when not focusing. I think it will be helpful.

Again, great job!!

vanelnw commented:
hello @pedroelton, 
Thanks for your efforts redesigning this feature, I agreer with suggestions pointed by @vitorsaa2k and I would also add suggestions about the links field : 
1- I think it would be better to check if the input is a correct link
2- The input should be reset after adding a link

https://user-images.githubusercontent.com/38879488/219535386-55107a61-73c0-477d-bfb8-3e1516b54c64.mp4

Lucas-E reviewed (commented):
Hey Pedro. First of all, awesome work here. When i was testing the changes and checked the hours, the red text shows _Hours - Best-case < Hours - Most-case < Hours - Most-case_
Should it show "Hours - Best-case < Hours - Most-case < Hours - Worst-case" instead?
![PedroPRFirst](https://user-images.githubusercontent.com/92558269/219882541-ba76cc77-8b9d-4a40-9454-6c22de113c6c.png)

angelinatr15 commented:
Hello @pedroelton,

The padding on the buttons look great! 

The radio buttons also look great compared to the previous drop downs. 

The "Hours" section also looks alot more organized. I think it would look nicer if the font was the same size as the other sections. 

I was able to create a new task. When I tried to edit this task after creation, the new resource component is not part of the edit component. Should we make it appear here also? Please see image below for reference to the edit component. 


<img width="398" alt="Screenshot 2023-02-18 at 2 19 20 PM" src="https://user-images.githubusercontent.com/97488347/219902133-75c7134c-256a-4a2c-8c1b-affb3bfcfbb9.png">

ayush0904 commented:
Hello @pedroelton 

Superb work, both in terms of functionality and user interface.
I'd appreciate some comments, and since this is my first PR for the project, if I'm mistaken, kindly correct me. 

1 ) Please see the below image, in the resource section I was able to add the same resource more than 1 time. I am not sure whether this is a functionality or a bug. 
<img width="466" alt="Screen Shot 2023-02-18 at 2 08 50 PM" src="https://user-images.githubusercontent.com/15180550/219905857-b7a92f53-4221-45f9-9212-f688f564519f.png">

**Solution** : We can maintain 2 list and can implement the pop and insert functionality between the 2

2) I was able to submit a task with the end date before the start date. I believe this should not be permitted
<img width="473" alt="Screen Shot 2023-02-18 at 4 26 11 PM" src="https://user-images.githubusercontent.com/15180550/219905968-e1319405-4da5-44d6-858d-12a07bc4dc1d.png">

3) After submitting a task, when we edit it again. The time field format is different. It is using Date/Time. However, while submitting a task the field is Date only.
<img width="469" alt="Screen Shot 2023-02-18 at 4 28 50 PM" src="https://user-images.githubusercontent.com/15180550/219906000-b665b9ab-e804-490b-bf6f-f57db57e8d4c.png">

4) I am unaware of the formula used for finding the Best and Worst Case time. However, for both the 1 and 1.5 in Best Case the Worst Case is coming out as 2
<img width="340" alt="Screen Shot 2023-02-18 at 4 25 54 PM" src="https://user-images.githubusercontent.com/15180550/219906027-29b24289-74e3-4a05-8660-eaf214e4fdbf.png">

5) The "Hours" has a different Font-Family and Font-Weight as compared to "Status" and other fields. 
<img width="130" alt="Screen Shot 2023-02-18 at 4 51 18 PM" src="https://user-images.githubusercontent.com/15180550/219906135-3b1cb0ac-89da-4655-9fe0-c16eb23a1cf4.png">

7) Once we select and add a field, I believe the field should be cleared. Please see the below image, I typed "dede" and clicked the "+" button, the result is shown in the drop section. However, the field is not cleared. The same issue can be replicated for Resources also. 
<img width="468" alt="Screen Shot 2023-02-18 at 5 04 56 PM" src="https://user-images.githubusercontent.com/15180550/219906518-40b4626d-038d-4204-8cdc-9f8252f0b4d1.png">


**UI/UX Suggestions:**

I believe this is not the part of PR request. 

1) We can save alot of space while designing the Pop-up "Add Task". I believe we can increase the width of the pop-up box and can reduce the height so that a user does not need to scroll down while submitting the form. 
2) Fields like Hours and Start and End date can be created in a parallel grid.
3) Rather than hiding and showing the Save button, we can do Enable and Disable

Lunaliar reviewed (commented):
Hey, @pedroelton I tested out your changes and I think this is great progress! I have a few suggestions that might help the flow in the TagSearch.jsx file. I think we can use find instead of map in this instance to isolate the specific member we want to add to the project. Also, I don't think line 51 is actually doing anything functional, it's pointing to the value of the list items and not the input (which is what I'm guessing you were aiming for)
Here's my recommendation for the handleClick function:
![find recommendation](https://user-images.githubusercontent.com/98275089/220739309-3b42ba50-8c8b-4d27-a3cd-478b800b8435.png)

pedroelton commented:
Hi Nyaj, thank you for your feedback. I fixed it. I wanted to give more emphasis because the label for hours is on the right size, but the visual hierarchy was compromised with this over-emphasized title. It changed on my last commit.

> Hi @pedroelton,
> 
> Thanks for your efforts on this amazing redesign work, it makes this feature more functional and understandable for users!
> 
> Here I have some advice which I believe could make this work even better:
> 
> 1. A tiny issue, is it better for the format of "Hours"(red-circled) to be aligned with the other titles(green-underlined)?
>    ![截屏2023-02-07 下午1 04 00](https://user-images.githubusercontent.com/77643813/217365879-411528a1-e091-4224-895f-6d59997b4b1e.png)
> 
> 2.00:00 - 02:36 of the video: The selection logic of the "Resources" part works as a dropdown list, but the implementation is not, which is little bit confusing and weird. Is it possible for us to make it a real dropdown list?
> 
> 3.02:36 - 03:12 of the video: Is it possible for the old panel to be updated to the new version?
> 
> 4.03:12 - 04:10 of the video: The "Copy" function does not work with so many errors, could you pls take a look at it?

pedroelton commented:
Thanks for your feedback... The start and end date fields were not part of this PR, only the search filter.
I'm adding these new bugs to the bugs file. Especially the 1st one, since it's also strange when we delete those blue badges, it doubles the round icons. It was a later implementation as well.

> Hello @pedroelton
> 
> Superb work, both in terms of functionality and user interface. I'd appreciate some comments, and since this is my first PR for the project, if I'm mistaken, kindly correct me.
> 
> 1 ) Please see the below image, in the resource section I was able to add the same resource more than 1 time. I am not sure whether this is a functionality or a bug. <img alt="Screen Shot 2023-02-18 at 2 08 50 PM" width="466" src="https://user-images.githubusercontent.com/15180550/219905857-b7a92f53-4221-45f9-9212-f688f564519f.png">
> 
> **Solution** : We can maintain 2 list and can implement the pop and insert functionality between the 2
> 
> 2. I was able to submit a task with the end date before the start date. I believe this should not be permitted
> 
> <img alt="Screen Shot 2023-02-18 at 4 26 11 PM" width="473" src="https://user-images.githubusercontent.com/15180550/219905968-e1319405-4da5-44d6-858d-12a07bc4dc1d.png">
> 
> 3. After submitting a task, when we edit it again. The time field format is different. It is using Date/Time. However, while submitting a task the field is Date only.
> 
> <img alt="Screen Shot 2023-02-18 at 4 28 50 PM" width="469" src="https://user-images.githubusercontent.com/15180550/219906000-b665b9ab-e804-490b-bf6f-f57db57e8d4c.png">
> 
> 4. I am unaware of the formula used for finding the Best and Worst Case time. However, for both the 1 and 1.5 in Best Case the Worst Case is coming out as 2
> 
> <img alt="Screen Shot 2023-02-18 at 4 25 54 PM" width="340" src="https://user-images.githubusercontent.com/15180550/219906027-29b24289-74e3-4a05-8660-eaf214e4fdbf.png">
> 
> 5. The "Hours" has a different Font-Family and Font-Weight as compared to "Status" and other fields.
> 
> <img alt="Screen Shot 2023-02-18 at 4 51 18 PM" width="130" src="https://user-images.githubusercontent.com/15180550/219906135-3b1cb0ac-89da-4655-9fe0-c16eb23a1cf4.png">
> 
> 7. Once we select and add a field, I believe the field should be cleared. Please see the below image, I typed "dede" and clicked the "+" button, the result is shown in the drop section. However, the field is not cleared. The same issue can be replicated for Resources also.
> 
> <img alt="Screen Shot 2023-02-18 at 5 04 56 PM" width="468" src="https://user-images.githubusercontent.com/15180550/219906518-40b4626d-038d-4204-8cdc-9f8252f0b4d1.png">
> 
> **UI/UX Suggestions:**
> 
> I believe this is not the part of PR request.
> 
> 1. We can save alot of space while designing the Pop-up "Add Task". I believe we can increase the width of the pop-up box and can reduce the height so that a user does not need to scroll down while submitting the form.
> 2. Fields like Hours and Start and End date can be created in a parallel grid.
> 3. Rather than hiding and showing the Save button, we can do Enable and Disable

pedroelton commented:
Hi Angelina, thank you for your feedback. I fixed it on my last commit/push.
The idea was to keep it as a title not as a label, the label would be the hours. But after your feedback I realized that the visual hierarchy made it over-emphasised, and that's a design issue for sure. Looks better now.
:) 
> Hello @pedroelton,
> 
> The padding on the buttons look great!
> 
> The radio buttons also look great compared to the previous drop downs.
> 
> The "Hours" section also looks alot more organized. I think it would look nicer if the font was the same size as the other sections.
> 
> I was able to create a new task. When I tried to edit this task after creation, the new resource component is not part of the edit component. Should we make it appear here also? Please see image below for reference to the edit component.
> 
> <img alt="Screenshot 2023-02-18 at 2 19 20 PM" width="398" src="https://user-images.githubusercontent.com/97488347/219902133-75c7134c-256a-4a2c-8c1b-affb3bfcfbb9.png">

pedroelton commented:
Hi Vanel, thank you for your feedback. But this is out of the scope of this PR. The change was only the resources field and the redesign/reorganized the Hours component. :)

> hello @pedroelton, Thanks for your efforts redesigning this feature, I agreer with suggestions pointed by @vitorsaa2k and I would also add suggestions about the links field : 1- I think it would be better to check if the input is a correct link 2- The input should be reset after adding a link

pedroelton commented:
Thank you for your feedback Prabhjot! That's very interesting, I didn't realize this issue on the filter. I'm addressing this to the dev that will take care of this component. 
> Hey @pedroelton, Filter is not working properly please do the required changes and I will approve this.
> 
> Please check the video

pedroelton commented:
Thank you for your feedback. Items 1 to 3 improve the UX dramatically. I'm addressing this issues to the dev that will take care of this component now.

Thank you, Vitor.

> vitorsaa2k



> Hello @pedroelton, i have some suggestions about the UX of the TagsSearch component
> 
> I think it should work like showed in the video below:
> 
> 1. Make the dropdown disapear when the input is not on focus and appear again when focused.
> 2. Make the input value controlled.
> 3. Make the input value be reseted to an empty string when a name is selected.
> 
> This last suggestion is just a "if you wanna add" differently from the other ones which are "must have"
> 
> 4. Also, the filter could search from the first letters to the last letters of each name, for example: when you search for "pe" it should show the names that starts with "pe" for example "pedro". Will be needed the build a better algorithm to deal with this searching, but it should not be too hard.

one-community commented:
Everyone please approve this if your suggestions have been integrated.

pedroelton commented:
Thank you Jae, I'm passing this PR to Raul, in case it needs more changes and improvements.

> Everyone please approve this if your suggestions have been integrated.

agujjar2206 commented:
Hello @pedroelton, I have tested the changes you mentioned in the video and everything seems to work well.

<img width="505" alt="672-1" src="https://user-images.githubusercontent.com/94432002/221333390-3e1e74db-ef94-4da0-b0d1-84cd54a01a65.png">

Thank you.
